File: example/python/picsocket_receiver.py
# Copyright (C) 2024 Coder.AN
# Email: an.hongjun@foxmail.com
# Page: www.anhongjun.top

import sys
import cv2
import picsocket


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: {} <listen-port>".format(sys.argv[0]))
    img_receiver = picsocket.ImgReceiver(int(sys.argv[1]))

    while True:
        img = img_receiver.read()
        cv2.imshow("receiver", img)
        cv2.waitKey(5)

File: example/picsocket_receiver.cpp
/*
 * Copyright (C) 2024 Coder.AN
 * Email: an.hongjun@foxmail.com
 * Page: www.anhongjun.top
 */
 #include <cstdlib>
 #include <cstdio>
 #include "pic_socket.h"

 int main(int argc, char* argv[])
 {
    if (argc != 2){
        printf("Usage: %s <listen-port>\n", argv[0]);
        return -1;
    }

    UDPImgReceiver img_receiver(atoi(argv[1]));

    while (true)
	{
		cv::Mat frame = img_receiver.read();
		cv::imshow("receiver", frame);
		cv::waitKey(30);
	}
	return 0;
 }